src/createContext.ts

```diff
@@ -94,12 +94,26 @@ const defineSymbol = (context: Context, name: string, value: Value) => {
 // If the builtin is a function, wrap it such that its toString hides the implementation
 export const defineBuiltin = (context: Context, name: string, value: Value) => {
   if (typeof value === 'function') {
-    const wrapped = (...args: any) => value(...args)
-    const funName = name.split('(')[0].trim()
-    const repr = `function ${name} {\n\t[implementation hidden]\n}`
-    wrapped.toString = () => repr
 
-    defineSymbol(context, funName, wrapped)
+    if (value.hasOwnProperty('isThunkAware')){
+      const wrapped = Object.assign(
+        (...args: any) => value(...args),
+        {isThunkAware : value.isThunkAware}
+      )
+      const funName = name.split('(')[0].trim()
+      const repr = `function ${name} {\n\t[implementation hidden]\n}`
+      wrapped.toString = () => repr
+      defineSymbol(context, funName, wrapped)
+    }
+    else{
+      const wrapped = (...args: any) => value(...args)
+      const funName = name.split('(')[0].trim()
+      const repr = `function ${name} {\n\t[implementation hidden]\n}`
+      wrapped.toString = () => repr
+
+      defineSymbol(context, funName, wrapped)
+    }
+
```

Comment:
A bit lengthy but it works.
You can simply write:
```
if (value.hasOwnProperty('isThunkAware')){
  wrapped.isThunkAware = value.isThunkAware;
}
```

TypeScript might shows some errors, but you can solve them by either adding a comment `// @ts-ignore` before the error line, or use:

```
      let wrapped = ... // replace "const wrapped = ..."
      wrapped = Object.assign(
        wrapped,
        {isThunkAware : value.isThunkAware}
      )
```

Reply:
This part is modified